File: example-package/src/tracing.ts
import dotenv from 'dotenv';
dotenv.config();
const { Resource } = require('@opentelemetry/resources');

import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
import { NodeSDK } from '@opentelemetry/sdk-node';
import * as otel from '@opentelemetry/auto-instrumentations-node';
import { ConsoleSpanExporter } from '@opentelemetry/tracing';
import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-proto';
import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';

export const provider = new NodeTracerProvider();

const sdk = new NodeSDK({
    resource: new Resource({
        [SemanticResourceAttributes.SERVICE_NAME]: 'service-name-here',
    }),
    traceExporter: new OTLPTraceExporter(),
    instrumentations: [
        otel.getNodeAutoInstrumentations()
    ] 
});

sdk.start()

File: api-example2/tracing.py
from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import (
    BatchSpanProcessor,
    ConsoleSpanExporter,
)
import os
from opentelemetry.sdk.resources import Resource
from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
from opentelemetry.propagators.b3 import B3Format

def initialize_tracer():
    # Create a Resource instance
    resource = Resource.create({
        'service.name': 'service-name-here'
    })

    # Create a TracerProvider instance
    provider = TracerProvider(resource=resource)
    #propagator = B3Format()

    # Create an OTLP SpanExporter and BatchSpanProcessor
    span_exporter = OTLPSpanExporter(endpoint='http://0.0.0.0:4317')
    processor = BatchSpanProcessor(span_exporter)

    # Add the processor to the TracerProvider
    provider.add_span_processor(processor)

    # Register the provider as the global tracer provider
    trace.set_tracer_provider(provider)


def get_tracer(service_name):
    return trace.get_tracer(service_name)